packages/medusa/src/interfaces/price-selection-strategy.ts

```diff
@@ -7,23 +7,29 @@ export interface IPriceSelectionStrategy {
   /**
    * Instantiate a new price selection strategy with the active transaction in
    * order to ensure reads are accurate.
-   * @param manager EntityManager with the queryrunner of the active transaction
+   * @param manager EntityManager with the query runner of the active transaction
    * @returns a new price selection strategy
    */
   withTransaction(manager: EntityManager): IPriceSelectionStrategy
 
   /**
    * Calculate the original and discount price for a given variant in a set of
    * circumstances described in the context.
-   * @param variant The variant id of the variant for which to retrieve prices
+   * @param variantId The variant id of the variant for which to retrieve prices
    * @param context Details relevant to determine the correct pricing of the variant
    * @return pricing details in an object containing the calculated lowest price,
    * the default price an all valid prices for the given variant
    */
   calculateVariantPrice(
-    variant_id: string,
+    variantId: string,
     context: PriceSelectionContext
   ): Promise<PriceSelectionResult>
+
+  /**
+   * Notify price selection strategy that variants prices have been updated.
+   * @param variantIds The ids of the updated variants
+   */
+  onVariantsPricesUpdate(variantIds: string[]): Promise<void>
```

Comment:
thoughts: I am fine with keeping it as is for now. But, we need to think of a more reliable and non tighten approach for the hook mechanism. We could have a hook service that works like the event bus except that we await the subscribers as a blocking step before the flow continue. Let's discuss that point at a later time but I just wanted to mention it. wdyt? In that case we could emit a `onVariantUpdate` pass the updated fields and the user would have the ability to hook himself into the flow through those hooks